### PR TITLE
API improvements

### PR DESCRIPTION
**Pull request by Jarlakxen**
Hi! In this PR I worked on improving the return types in the API specification, and also added more integration tests. The IT are not going increase the coverage very much, but I think they are useful to see examples of the API in action.
Another small thing, that I did, was a change in the onEvent method in the api/connection, to return a function to remove the listener from the event emitter.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "test:coverage": "node node_modules/istanbul/lib/cli.js cover -e .ts  -x \"*.d.ts\" -x \"*.spec.ts\" node_modules/jasmine/bin/jasmine.js -- JASMINE_CONFIG_PATH=jasmine.json",
     "lint": "tslint --format stylish --force src/**/*.ts",
     "watch": "gulp watch",
-    "clean": "gulp clean"
+    "clean": "rm ./release -rf"
```
Comment (hedefalk) [inline package.json]: I think the reason I used rimraf was to be able to run tests on Windows. Dunno state of things now…?

Comment (Jarlakxen) [reply package.json]: Oh! Good point. My fault, I didn't consider windows.

Comment (Jarlakxen) [reply package.json]: I fixed it

**Comment by Jarlakxen** [comment]
Hey @hedefalk! Any updates on this? =)

**Comment by hedefalk** [comment]
Sorry for being slow, been swamped with other stuff.

Seems good! Are you using any clients against it? It was something that didn't seem backwards compatible iirc? Not that atom/vscode always have to work against master, but maybe we need some kind of versioning branches otherwise. Was going to look into it but got busy…

**Comment by hedefalk** [comment]
@Jarlakxen ^

**Comment by Jarlakxen** [comment]
Hey! No problem =)

I'm not using this lib in any client, right now. But I'm a huge fun of Atom, so I want to start working on that. I started with this library because I wanted to understand how Ensime works =)

I think I didn't break backwards compatibility, but there are some changes in the methods signature, due to the improvements in the types definitions. This is not going to be a problem with CoffeeScript, but maybe it will generate some compilation erros/warnings in VSCode.

BTW, I sent you a message in gitter.

**Comment by hedefalk** [comment]
Oh, jesus. I have much to read up there. Was it long ago? :)

I need to release a working Atom for 2.0 it seems and the new .ensime format.

**Comment by Jarlakxen** [comment]
I think I sended the message like a month ago =P
What's different in 2.0? The IT are running using 2.0.0-M1, I think.

**Pull request merged**